src/lib/service/__tests__/walletConnectService.test.ts

```diff
@@ -0,0 +1,282 @@
+import { expect } from '@jest/globals';
+
+import {
+    DisconnectError,
+    MakeRequestError,
+    NoConnectionError,
+    NoSessionError,
+    NotRunningError,
+} from '../../../lib/errors/WalletConnectErrors';
+import { StellarNetwork } from '../../../lib/stellar/StellarNetwork';
+import { WCClient } from '../lib/walletconnect';
+import {
+    IWalletConnectRequestParams,
+    IWalletConnetConnectionParams,
+    WalletConnectAllowedMethods,
+    WalletConnectNetwork,
+    WalletConnectService,
+    WalletConnectTargetChain,
+} from '../walletConnect';
+
+const NO_CONNECTION_ERROR = new NoConnectionError();
+const DISCCONECT_ERROR = new DisconnectError();
+const NO_SESSION_ERROR = new NoSessionError();
+const MAKE_REQUEST_ERROR = new MakeRequestError();
+const NOT_RUNNING_ERROR = new NotRunningError();
+
+const mockClientMethods = {
+    init: jest.fn(),
+    connect: jest.fn(),
+    disconnect: jest.fn(),
+    request: jest.fn(),
+    session: { getAll: jest.fn() },
+};
+
+const mockModalMethods = {
+    openModal: jest.fn(),
+    closeModal: jest.fn(),
+};
+
+jest.mock('../../../constants', () => ({
+    PROJECT_ID_FOR_WALLET_CONNECT: 'exampleId',
+    DAPP_BASE_URL: 'exampleUrl',
+    HORIZON_NETWORK_PASSPHRASE: 'Test SDF Network ; September 2015',
+}));
+
+jest.mock('../lib/walletconnect', () => {
+    return {
+        WCClient: {
+            init: () => mockClientMethods,
+        },
+        WCModal: jest.fn().mockImplementation(() => {
+            return mockModalMethods;
+        }),
+    };
+});
+
+describe('Wallet Connect Service', () => {
+    let service: WalletConnectService;
+    const ERROR = new Error();
+    const consoleError = jest.spyOn(console, 'error').mockImplementation();
+    const SESSION_ID = 'test';
+    const sessions = [{ topic: SESSION_ID }];
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        service = new WalletConnectService();
+    });
+
+    describe('createClient', () => {
+        it('should set and return the client', async () => {
```

Comment:
Remember to maintain consistency with how we declare our tests. This `it` message has to be capitalized.